ambry-account/src/main/java/com/github/ambry/account/RouterStore.java

```diff
@@ -0,0 +1,345 @@
+    private final Collection<Account> accounts;
+    private final Pair<String, Path> backupPrefixAndPath;
+    private Map<String, String> potentialNewState;
+    private String newBlobID = null;
+
+    /**
+     * @param accounts The {@link Account}s to update.
+     */
+    ZKUpdater(Collection<Account> accounts) {
+      this.accounts = accounts;
+      Pair<String, Path> backupPrefixAndPath = null;
+      try {
+        backupPrefixAndPath = backup.reserveBackupFile();
+      } catch (IOException e) {
+        logger.error("Error reserving backup file", e);
+      }
+      this.backupPrefixAndPath = backupPrefixAndPath;
+    }
+
+    @Override
+    public ZNRecord update(ZNRecord znRecord) {
+      // There are several steps to finish an update
+      // 1. Fetch the list from the ZNRecord
+      // 2. Fetch the AccountMetadata from the blob id if the list exist in the ZNRecord
+      // 3. Construct a new AccountMetadata
+      // 4. save it as a blob in the ambry server
+      // 5. Add the new blob id back to the list.
+
+      // Start step 1:
+      ZNRecord recordToUpdate;
+      if (znRecord == null) {
+        logger.debug(
+            "ZNRecord does not exist on path={} in HelixPropertyStore when updating accounts. Creating a new ZNRecord.",
+            ACCOUNT_METADATA_BLOB_IDS_PATH);
+        recordToUpdate = new ZNRecord(ZN_RECORD_ID);
+      } else {
+        recordToUpdate = znRecord;
+      }
+
+      String errorMessage = null;
+      List<String> accountBlobIDs = recordToUpdate.getListField(ACCOUNT_METADATA_BLOB_IDS_LIST_KEY);
+      int newVersion = 1;
+      Map<String, String> accountMap = null;
+      if (accountBlobIDs != null && accountBlobIDs.size() != 0) {
+        // parse the json string list and get the blob id with the latest version
+        try {
+          BlobIDAndVersion blobIDAndVersion = null;
+          for (String accountBlobIDInJson : accountBlobIDs) {
+            BlobIDAndVersion current = BlobIDAndVersion.fromJson(accountBlobIDInJson);
+            if (blobIDAndVersion == null || blobIDAndVersion.version < current.version) {
+              blobIDAndVersion = current;
+            }
+          }
+          newVersion = blobIDAndVersion.version + 1;
+
+          // Start Step 2:
+          accountMap = readAccountMetadataFromBlobID(blobIDAndVersion.blobID);
+          // make this list mutable
+          accountBlobIDs = new ArrayList<>(accountBlobIDs);
+        } catch (JSONException e) {
+          accountServiceMetrics.remoteDataCorruptionErrorCount.inc();
+          errorMessage = String.format("Exception occurred when parsing the blob id list from {}", accountBlobIDs);
+          logger.error(errorMessage);
+          throw new IllegalStateException(errorMessage, e);
+        } catch (Exception e) {
+          errorMessage =
+              String.format("Unexpected exception occurred when parsing the blob id list from {}", accountBlobIDs);
```

Comment:
if you choose to String.format, can you change it to following style (basically replace `{}` with `%s`):
```
String.format("Unexpected exception occurred when parsing the blob id list from %s", accountBlobIDs);
```
Make change to all occurrences. This can pass intellij inspection.

Reply:
figured that string concatenation would just do the work.